{
  try {
    XQQueryException ex=new XQQueryException("Hello world!",new QName("foo"));
    junit.framework.Assert.assertEquals("A-XQQEX-2.1: Retrieve error code from an XQQueryException",new QName("foo"),ex.getErrorCode());
  }
 catch (  Exception e) {
    junit.framework.Assert.fail("A-XQQEX-2.1: Retrieve error code from an XQQueryException failed with message: " + e.getMessage());
  }
  try {
    XQQueryException ex=new XQQueryException("Hello world!");
    junit.framework.Assert.assertEquals("A-XQQEX-2.2: Error code is null when not available",null,ex.getErrorCode());
  }
 catch (  Exception e) {
    junit.framework.Assert.fail("A-XQQEX-2.2: Retrieve error code from an XQQueryException failed with message: " + e.getMessage());
  }
}
